### PR TITLE
Make it possible to disable  tapping to dismiss a tooltip.

### PR DESCRIPTION
**Pull request by hannah-hyj**
issue: https://github.com/flutter/flutter/issues/137438

Use case: I want to add action button is tooltip and thus want to disable tap to dismiss a tooltip.

image

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/tooltip.dart [hunk 1]
```diff
@@ -290,6 +291,11 @@ class Tooltip extends StatefulWidget {
   /// for mouse pointer exits the widget.
   final Duration? showDuration;
 
+  /// The length of the delay time before dismissing tooltips by tap.
+  ///
+  /// Defaults to 0 milliseconds (tooltips are dismissed immediately).
+  final Duration? tapToDismissDelayDuration;
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/tooltip.dart:297]: This feels like a weird api, is the goal to be able to have other gesture recognizer on the tool tip to fire? what if the gesture take longer than the duration to fire?  for example if someone add a long press gesture recognizer in tooltip.

Should we just have a flag to turn off tap dismissal?

Comment (hangyujin) [reply packages/flutter/lib/src/material/tooltip.dart]: make sense, I changed it to a flag for tap dismissal

File: packages/flutter/lib/src/material/tooltip.dart [hunk 2]
```diff
@@ -290,6 +291,11 @@ class Tooltip extends StatefulWidget {
   /// for mouse pointer exits the widget.
   final Duration? showDuration;
 
+  /// Whether the tooltip should be dismissed by tap.
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/tooltip.dart:294]:
```suggestion
  /// Whether the tooltip can be dismissed by tap.
```

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**